src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopyColumnOrderHintCollection.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Data.SqlClient
+{
+    /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/SqlBulkCopyColumnOrderHintCollection/*'/>
+    public sealed class SqlBulkCopyColumnOrderHintCollection : CollectionBase
+    {
+        private readonly HashSet<string> _columnNames = new HashSet<string>();
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Item/*'/>
+        public SqlBulkCopyColumnOrderHint this[int index] => (SqlBulkCopyColumnOrderHint)List[index];
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlBulkCopyColumnOrderHintCollection.xml' path='docs/members[@name="SqlBulkCopyColumnOrderHintCollection"]/Add[@name="columnOrderHintParameter"]/*'/>
+        public SqlBulkCopyColumnOrderHint Add(SqlBulkCopyColumnOrderHint columnOrderHint)
+        {
+            if (columnOrderHint == null)
+            {
+                throw new ArgumentNullException(nameof(columnOrderHint));
+            }
+            if (string.IsNullOrEmpty(columnOrderHint.Column) ||
+                columnOrderHint.SortOrder == SortOrder.Unspecified)
+            {
+                throw SQL.BulkLoadInvalidOrderHint();
```

Comment:
Since these public APIs throw exceptions, it also must be part of documentation.
Please look at existing documentation (e.g. [SqlNotificationRequest.xml](https://github.com/dotnet/SqlClient/blob/master/doc/snippets/Microsoft.Data.Sql/SqlNotificationRequest.xml#L41-L42)) and apply same for these new exceptions.